src/Cardano/Wallet/Api/Types/WalletBalance.hs

```diff
@@ -0,0 +1,24 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+module Cardano.Wallet.Api.Types.WalletBalance where
+
+import Prelude
+
+import Cardano.Wallet.Api.JSON
+    ( defaultRecordTypeOptions )
+import Data.Aeson
+    ( FromJSON (..), ToJSON (..), genericParseJSON, genericToJSON )
+import GHC.Generics
+    ( Generic )
+
+import qualified Cardano.Wallet.Api.Types.Amount as T
+
+data WalletBalance = WalletBalance
+    { _available :: !T.Amount
+    , _total :: !T.Amount
```

Comment:
units are missing from the amounts here :+1:

Reply:
The unit is defined as part of the definition of `Amount`:
```hs
data Amount = Amount
    { _quantity :: Natural
    , _unit :: T.CurrencyUnit
    } deriving (Eq, Generic, Show)

instance FromJSON Amount where
    parseJSON = genericParseJSON defaultRecordTypeOptions
instance ToJSON Amount where
    toJSON = genericToJSON defaultRecordTypeOptions
```
This seems consistent with the Swagger specification. Perhaps I have misunderstood your meaning?